src/include/sof/audio/module_adapter/library/module_api_ver.h

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+ */
+
+#ifndef __MODULE_API_VER_H__
+#define __MODULE_API_VER_H__
+
+/*
+ *  Api version 5.0.0 for sof loadable modules
+ */
+
+#define SOF_MODULE_API_MAJOR_VERSION 5
+#define SOF_MODULE_API_MIDDLE_VERSION 0
+#define SOF_MODULE_API_MINOR_VERSION 0
+
+typedef union sof_module_api_version {
+	uint32_t full;
+	struct {
+		uint32_t minor    : 10;
+		uint32_t middle   : 10;
+		uint32_t major    : 10;
+		uint32_t reserved : 2;
+	} fields;
+} sof_module_api_version;
+
+typedef const struct {
+	uint32_t format;
+	sof_module_api_version api_version_number;
+} sof_module_api_build_info;
```

Comment:
That struct is the same in IADK modules and it is not packed https://github.com/thesofproject/converged-sof-modules/blob/main/FW/src/intel_adsp/include/adsp_stddef.h#L61 does packing it to fixed size won't harm anything?

Reply:
@pjdobrowolski adding a packed attribute might or might not break it. Say, if you had
```
struct x {
    uint16_t s;
    uint32_t i;
};
```
without packed, then "usually" compilers would align `i` on a 32-bit boundary and thus leave a 16-bit gap in the structure. If this is a part of an API and, say, is sent between the host and the DSP, then 2 different compilers are used. If they both happen to apply the same alignment in this case, it will work. But it's unreliable and it might change, at which point it will break. Now if you did have this case and it did work and now you add a packed attribute, that gap will disappear, and it will break the ABI because it will change the layout of the structure.

Reply:
For that point it is working and I wouldn't change anything about that.

Reply:
I would check how the compiler(s) pack it now, i.e. if there are any gaps. Probably there aren't, then I'd add the attribute to make sure it stays that way. If there are gaps (unlikely), I'd add "reserved" fields there and also a packed attribute to also make the structure immutable. But again - all that is needed only if it's indeed a part of the ABI

Reply:
@lyakh your are testing loadable modules now, can you check if it is necessary? It is just one macro and I would like to switch into dynamic memory allocation for loadable modules now.